store/store.md

```diff
@@ -1 +1,60 @@
 # Store
+
+## Abstract
+
+The Store interface defines methods for storing and retrieving blocks, commits, and the state of the blockchain.
+
+## Protocol/Component Description
+
+The Store interface defines the following methods:
+
+- `Height`: Returns the height of the highest block in the store.
+- `SetHeight`: Sets the height saved in the Store if it is higher than the existing height.
+- `SaveBlock`: Saves a block along with its seen commit.
+- `LoadBlock`: Returns a block at a given height.
+- `LoadBlockByHash`: Returns a block with a given block header hash.
+- `SaveBlockResponses`: Saves block responses in the Store.
+- `LoadBlockResponses`: Returns block results at a given height.
+- `LoadCommit`: Returns a commit for a block at a given height.
+- `LoadCommitByHash`: Returns a commit for a block with a given block header hash.
+- `UpdateState`: Updates the state saved in the Store. Only one State is stored.
+- `LoadState`: Returns the last state saved with UpdateState.
+- `SaveValidators`: Saves the validator set at a given height.
+- `LoadValidators`: Returns the validator set at a given height.
+
+An implementation of the Store interface is the `DefaultStore` struct which uses a `TxnDatastore` interface for its underlying storage. The `TxnDatastore` interface has two key-value store implementations:
+
+- `NewDefaultInMemoryKVStore`: Builds a key-value store that uses the BadgerDB library and operates in-memory, without accessing the disk. Used only across unit tests and integration tests.
```

Comment:
[non-blocking] follow up: we should move/rename this to `NewTestKVStore` then if it is only used in testing.

Reply:
Follow up issue: https://github.com/rollkit/rollkit/issues/1278